src/Microsoft.AspNet.Mvc.Core/ModelBinding/FormCollectionModelBinder.cs

```diff
@@ -30,19 +29,11 @@ public class FormCollectionModelBinder : IModelBinder
             if (request.HasFormContentType)
             {
                 var form = await request.ReadFormAsync();
-                if (bindingContext.ModelType.GetTypeInfo().IsAssignableFrom(form.GetType().GetTypeInfo()))
```

Comment:
We specifically only support using `IFormCollection` here. `FormCollection` is an 'internal' type and we don't encourage you to use it directly outside of tests.

I'd rather not build more features into this (activation of your custom type). There are far fewer useful things to do with custom types here than in other places in model binding.